File: www/cgi-bin/hello.pl
#!/usr/bin/perl
use strict;
use warnings;
use POSIX qw(strftime);

print "Content-Type: text/html\r\n\r\n";
print "<html><body>";
print "<h1>Hello from Perl CGI!</h1>";
print "<p>Time is: " . strftime("%Y-%m-%d %H:%M:%S", localtime) . "</p>";
print "</body></html>";

File: www/cgi-bin/hello.py
#!/usr/bin/python3
import os
import datetime

print("Content-Type: text/html\r\n\r\n")
print("<html><body>")
print("<h1>Hello from CGI!</h1>")
print(f"<p>Time is: {datetime.datetime.now()}</p>")
print("<h2>Environment Variables:</h2><ul>")
for k, v in os.environ.items():
    print(f"<li><b>{k}</b>: {v}</li>")
print("</ul></body></html>")
